paddle/fluid/operators/shape_op.cu

```diff
@@ -12,12 +12,58 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
+#include <cuda.h>
+#include <vector>
 #include "paddle/fluid/operators/shape_op.h"
 
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+using LoDTensor = framework::LoDTensor;
+using SelectedRows = framework::SelectedRows;
+
+template <typename T>
+class ShapeCUDAKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    auto* in_var = ctx.InputVar("Input");
+    framework::DDim in_dims;
+    if (in_var->IsType<SelectedRows>()) {
+      in_dims = in_var->Get<SelectedRows>().value().dims();
+    } else {
+      in_dims = in_var->Get<LoDTensor>().dims();
+    }
+
+    auto* out_t = ctx.Output<Tensor>("Out");
+    auto dim_size = in_dims.size();
+    out_t->Resize({dim_size});
+    // The data type in DDims is int64_t. Downcasting it into int32_t
```

Comment:
Why not output int64_t? And it's better to add some checks if you really want to make the output int32_t.  For example,

```
auto safe_cast = [](int64_t x){
  PADDLE_ENFORCE_LE(x, std::numeric_limits<int32_t>::max(),  err_msg);
  PADDLE_ENFORCE_GE(x, std::numeric_limits<int32_t>::min(),  err_msg);
  return static_cast<int32_t>(x);
}
in_dims_data.push_back(safe_cast(in_dims[i]));
```